pkg/api/api.go

```diff
@@ -162,6 +161,10 @@ func (a *API) RegisterRoutesWithPrefix(prefix string, handler http.Handler, auth
 func (a *API) newRoute(path string, handler http.Handler, isPrefix, auth, gzip bool, methods ...string) (route *mux.Route) {
 	if auth {
 		handler = a.AuthMiddleware.Wrap(handler)
+
+		// Assuming that we only need consistency controls when the request is for a specific tenant.
+		// Not all requests that require a tenant also support consistency, but it doesn't hurt if we propagate it anyway.
+		handler = querierapi.ConsistencyMiddleware().Wrap(handler)
```

Comment:
I think my comment is the opposite. Why not just always inject it? Also look at who else **reads** `AuthMiddleware` cause I think should inject it also when directly querying queriers for consistency.

Reply:
>Why not just always inject it?

I mean, why only when auth is used? We could just always enable it. Then the fact many APIs currently don't support it is a separate discussion, but for consistency we should always enable it.

Alternatively, we could move to the other side of the specturum and just enable it for querier's API, so only for routes registered by `RegisterQueryAPI()`.

I don't feel strong about this comment, but it still looks weird to bundle it to auth, given has nothing to do with auth.

Reply:
thanks for the review. Can you check out how dd76194ba5fa0d1d27e108d0a25f9b7dffa47f70 looks like?